lib/datasource/docker/index.ts

```diff
@@ -66,9 +66,14 @@ export function getRegistryRepository(
   registryUrl: string
 ): RegistryRepository {
   if (registryUrl !== defaultRegistryUrls[0]) {
-    const registry = registryUrl.replace('https://', '');
-    const registryEndingWithSlash = registry.replace(/\/?$/, '/');
+    const registryEndingWithSlash = registryUrl
+      .replace(/^https?:\/\//, '')
+      .replace(/\/?$/, '/');
     if (lookupName.startsWith(registryEndingWithSlash)) {
+      let registry = registryUrl.replace(/\/?$/, '');
```

Comment:
Maybe we should add a `trimTrailingSlash` util function too?

Reply:
Added :+1: